README.md

```diff
@@ -248,7 +208,201 @@ WEBPACKER_DEV_SERVER_HOST=0.0.0.0 ./bin/webpack-dev-server
+Webpacker gives you a default set of configuration files for test, development and
+production environments in `config/webpack/*.js`. You can configure each individual
+environment in their respective files or configure them all in the base
+`config/webpack/environment.js` file.
+
+By default, you don't need to make any changes to `config/webpack/*.js`
+files since it's all standard production-ready configuration. However,
+if you do need to customize or add a new loader, this is where you would go.
+
+Here is how you can modify webpack configuration:
+
+You might add separate files to keep your code more organized.
+
+```js
+// config/webpack/custom.js
+module.exports = {
+  resolve: {
+    alias: {
+      jquery: 'jquery/src/jquery',
+      vue: 'vue/dist/vue.js',
+      React: 'react',
+      ReactDOM: 'react-dom',
+      vue_resource: 'vue-resource/dist/vue-resource'
+    }
+  }
+}
+```
+
+Then `require` this file in your `config/webpack/base.js`:
+
+```js
+// config/webpack/base.js
+const { webpackConfig, merge } = require('@rails/webpacker')
+const customConfig = require('./custom')
+
+module.exports = merge(webpackConfig, customConfig)
+```
+
+If you need access to configs within Webpacker's configuration,
+you can import them like so:
+
+```js
+// config/webpack/base.js
+const { webpackConfig } = require('@rails/webpacker')
+
+console.log(webpackConfig.output_path)
+console.log(webpackConfig.source_path)
+```
+
+### Integrations
+
+Webpacker out of the box supports JS and static assets (fonts, images etc.)
+compilation. To enable support for Coffeescript or Typescript install
+relevant packages,
+
+**Coffeescript**
+
+```
+yarn add coffeescript coffee-loader
+```
+
+**Typescript**
+
+```
+yarn add typescript @babel/preset-typescript
+```
+
+Add tsconfig.json
+
+```json
+{
+  "compilerOptions": {
+    "declaration": false,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "lib": ["es6", "dom"],
+    "module": "es6",
+    "moduleResolution": "node",
+    "baseUrl": ".",
+    "paths": {
+      "*": ["node_modules/*", "app/javascript/*"]
+    },
+    "sourceMap": true,
+    "target": "es5",
+    "noEmit": true
+  },
+  "exclude": ["**/*.spec.ts", "node_modules", "vendor", "public"],
+  "compileOnSave": false
+}
+```
+
+#### CSS
+
+To enable CSS support in your application, add following packages,
+
+```
+yarn add css-loader mini-css-extract-plugin css-minimizer-webpack-plugin
+```
+
+optionally, add css extension to webpack config for easy resolution
+
+```js
+// config/webpack/base.js
+const { webpackConfig, merge } = require('@rails/webpacker')
+const customConfig = {
+  resolve: {
+    extensions: ['.css']
+  }
+}
+
+module.exports = merge(webpackConfig, customConfig)
+```
+
+To enable postcss, sass or less support, add css support first and
+then add the relevant pre-processors:
+
+#### Postcss
+
+```
+yarn add postcss-loader
+```
+
+#### Sass
+
+```
+yarn add sass-loader
```

Comment:
@gauravtiwari should it be `yarn add sass-loader --dev`?

https://github.com/webpack-contrib/sass-loader#getting-started

and for other loaders

Reply:
Yes, this has been a historical confusion. I think we can say that if you are building locally then use `--dev` otherwise it should be added with main dependencies: https://github.com/rails/webpacker/issues/117

Reply:
For this reason, the language here might be useful for the new docs: https://github.com/rails/webpacker/blob/fb9aa9622a7760cb6c21d304250e1c9b92b1a6ac/docs/yarn.md#add-an-npm-module-to-devdependencies

> Be careful not to add any build or app related JS modules in this fashion. Adding JS modules to `devDependencies` [will block them from being installed in **any** production environment](https://yarnpkg.com/lang/en/docs/cli/install/#toc-yarn-install-production-true-false).
>
> Docs from JS modules may instruct you to use `--dev` or `devDependencies`, but this is generally under the assumption that you are using a `node.js` workflow.